From b1ae882b40fe81c6dad50b6c5ff053bbc0b34058 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Wed, 4 Mar 2020 17:40:51 +0800
Subject: [PATCH] pinctrl/rockchip: get virq by irq_find_mapping

The rockchip pinctrl creates an irq domain by irq_domain_add_linear,
but it not means all pins' irq are in linear.

Such as rk808 interrupt, it defines in node:
	rk808 {
		interrupt-parent = <&gpio1>;
		interrupts = <21 IRQ_TYPE_LEVEL_LOW>;
	}

then i2c core driver will get virq by
	irq = of_irq_get(dev->of_node, 0);
		->irq_create_of_mapping(&oirq);

finally the rk808 irq is mapping in DOMAIN_BUS_WIRED irq domain.

It's better to get virq by irq_find_mapping.

Change-Id: Ib416ace5c2212c9a704b01e78f77d3425dc8d21b
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 drivers/pinctrl/pinctrl-rockchip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index 32276f227b71..ecc7b4dfb309 100644
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -3659,7 +3659,7 @@ static void rockchip_irq_demux(struct irq_desc *desc)
 
 		irq = __ffs(pend);
 		pend &= ~BIT(irq);
-		virq = irq_linear_revmap(bank->domain, irq);
+		virq = irq_find_mapping(bank->domain, irq);
 
 		if (!virq) {
 			dev_err(bank->drvdata->dev, "unmapped irq %d\n", irq);
-- 
2.35.3

